### ISSUE TITLE
Clans

### ISSUE THREAD

**Issue by Greenfoot5** [Q-Sharp/BattleBot#7]
Clans are a massive update to battle bot that brings a whole new set of commands to play with. It's already a wip, but now other areas need to be completed first. That's the json update #3 and options #6. I'll break it up a little.

- [ ] Anyone can create a clan. - Anyone at all. However, may want to add a minimum level requirement for this so that people actually know what they're doing. Each clan will have an id, which is how they're indexed in the main json dict. A key issue here is linking clans however.
- [ ] You can link the id of your BattleBot clan to your in-game clan. How this shall be done is currently unknown. One way would be to force people to their clan if when they create it along with clan name and tag. However, the other issue if verification. How can we confirm that their id is actually a correct id. Clanging ids could also mess up server links and profile links as well.
- [ ] You can join a clan! Users can join a clan. This means that users can link a clan onto their profile and it also display's their clan tag before their username when possible (and if they haven't manually set it).
- [ ] You can link a server and clan. This means that the server will be "linked" with the clan. For now, this just means that BattleBot will profile a few server details as well as an invite to the clan page. It will also automatically update the clan's icon if it hasn't been changed.
- [ ] Allow users to edit their clan. This is from a range of stats already defined in clans.json
- [ ] Allow users to promote/demote members according to the in-game ranks (but given better names).
- [ ] Advertisements. Anyone can setup an advertisement channel. This would mean that channel receives advertisements from clans that have an advertisement setup, and it would do so a periodic intervals. It would also be nice to improve this system to filter out "bad" advertisements, but I'll make a separate issue for that.
